editoast/src/views/projection.rs

```diff
@@ -337,6 +351,193 @@ pub async fn compute_projected_train_paths(
         .collect())
 }
 
+fn linear_interpolate(pos: &[u64], op_begin: u64, op_end: u64) -> Vec<u64> {
+    assert!(pos.len() > 1);
+    let total_dist = u64::abs_diff(op_begin, op_end);
+    let start = pos.first().unwrap();
+    let end = pos.last().unwrap();
+    pos.iter()
+        .map(|&x| ((x - start) / (end - start)) * total_dist + op_begin)
+        .collect()
+}
+
+fn global_interpolate(
+    pos: Vec<u64>,
+    op_map: HashMap<&String, u64>,
+    path_item_positions: &[u64],
+    path_item_idxs_and_op_ids: &[(usize, String)],
+) -> Vec<u64> {
+    assert!(pos.len() > 1);
+    let mut first_op = path_item_idxs_and_op_ids[0].clone();
+    let mut first_pos = path_item_positions[first_op.0];
+
+    let mut interpolated_pos: Vec<_> = vec![op_map[&first_op.1]];
+
+    for (idx, id) in path_item_idxs_and_op_ids.iter().skip(1) {
+        let next_pos = path_item_positions[*idx];
+        let current_positions: Vec<_> = pos
+            .clone()
+            .into_iter()
+            .filter(|position| *position > first_pos && *position <= next_pos)
+            .collect();
+        interpolated_pos.extend(linear_interpolate(
+            &current_positions,
+            op_map[&first_op.1],
+            op_map[id],
+        ));
+        first_pos = next_pos;
+        first_op = (*idx, id.clone());
+    }
+
+    interpolated_pos
+}
+
+#[allow(clippy::too_many_arguments)] // WARN
+pub async fn compute_projected_train_paths_op(
+    conn: &mut DbConnection,
+    core_client: Arc<CoreClient>,
+    valkey_client: Arc<ValkeyClient>,
+    infra: &Infra,
+    trains_schedules: Vec<models::TrainSchedule>,
```

Comment:
As preparation for paced trains, we should take schema instead of models.
```suggestion
    trains_schedules: Vec<schemas::TrainSchedule>,
```

Since we don't have the id anymore the output should be adapt to, use `Vec` instead of `HashMap`